ui/scripts/ui-custom/physicalResources.js

```diff
@@ -264,6 +264,94 @@
+                                </div>
+                                <div class="form-item">
+                                    <div class="name"><label>Sign-out URL</label></div>
+                                    <div class="value" style="width: ${valueWidth}px">${result.slourl}</div>
+                                </div>
+                                <div class="form-item">
+                                    <div class="name"><label>Contact Name</label></div>
+                                    <div class="value" style="width: ${valueWidth}px">${result.contactname}</div>
+                                </div>
+                                <div class="form-item">
+                                    <div class="name"><label>Contact Email</label></div>
+                                    <div class="value" style="width: ${valueWidth}px">${result.contactemail}</div>
+                                </div>
+                                <div>Raw Service Provider XML</div>
+                                <div style="height: 300px; overflow: scroll">
+                                    ${escapedXml}
+                                </div>
+                            `)
+                        ).dialog({
+                            title: 'SAML Service Provider Metadata',
+                            dialogClass: 'form-container',
+                            closeOnEscape: true,
+                            zIndex: 5000,
+                            width: dialogWidth,
+                            buttons: [{
+                                text: "Close",
+                                'class': 'cancel',
+                                click: function() {
+                                    $(this).dialog('destroy');
+                                    $('.hovered-elem').hide();
+                                }
+                            }, {
+                                text: "Renew Cert",
+                                click: function() {
+                                    let parentDialog = $(this);
+                                    cloudStack.dialog.confirm({message: "Are you sure you want to renew the SAML certificate?", action: function() {
+                                            $.ajax({
+                                                url: createURL('renewSPCertificates'),
+                                                dataType: "json",
+                                                success: function(json) {
+                                                    parentDialog.dialog('destroy');
+                                                }
+                                            })
+                                        }});
+                                }
+                            }]
+                        })
+                    },
+                    error: function(XMLHttpResponse) {
```

Comment:
Is the error response empty?

Reply:
Potentially not, I've added something to push an error to the user here if there is an error.